units/src/amount.rs

```diff
@@ -1755,7 +1755,10 @@ mod verification {
             if n1 >= 0 {
                 Ok(Amount::from_sat(n1.try_into().unwrap()))
             } else {
-                Err(ParseAmountError::Negative)
+                Err(OutOfRangeError {
+                    is_signed: true,
+                    is_greater_than_max: false
+                })
```

Comment:
This makes me happy with the change to use `is_negative`. If this had
```rust
is_signed: true,
is_negative: true,
```
I think I (as author) and reviewers would have thought "what does `is_signed` mean if `is_negative` is true?" And then caught the bug - do you agree @apoelstra or do you think this could have slipped through still?

Reply:
Agreed that in this case that `is_negative` would have been a clearer name.

But when `is_signed` is *false* it's not clear what `is_negative` is supposed to mean.

Really the correct solution is that we have two separate error types that have a differently named field.

Reply:
oh, this question shows its still not clear enough. `is_signed` is the signed-ness of the type (`Amount` vs `SignedAmount`) and `is_negative` is the signed-ness of the value (either being parsed or being converted). I was hoping that because the names are so similar in would force devs to read the docs on the error type - a far from perfect solution I know.

Reply:
My response is to ignore the names, interpret the error as a 4-valued enum, and try to avoid ever needing to interpret what the four values actually mean.

Reply:
That works ... if we write the tests correctly. lolz, lets leave this `amount` module and just not touch it or talk about it any more :) And I thought string parsing was supposed to be easy in Rust.